### PR TITLE
Add test cases for Reader 

### PR DESCRIPTION
**Pull request by dota17**
Add test cases for Reader in src\lib_json\json_reader.cpp.

### CODE REVIEW

File: src/test_lib_json/main.cpp [hunk 1]
```diff
@@ -2751,13 +2892,6 @@ JSONTEST_FIXTURE_LOCAL(ReaderTest, parseChineseWithOneError) {
   JSONTEST_ASSERT(reader.getFormattedErrorMessages() ==
                   "* Line 1, Column 19\n  Syntax error: value, object or array "
                   "expected.\n");
-  std::vector<Json::Reader::StructuredError> errors =
```
Comment (BillyDonahue) [inline src/test_lib_json/main.cpp]: why are these structured error verifications being deleted?

File: src/test_lib_json/main.cpp [hunk 2]
```diff
@@ -2653,16 +2653,126 @@ JSONTEST_FIXTURE_LOCAL(ReaderTest, parseWithNoErrors) {
...
+    bool ok = reader.parse("[ \"\\uD801\" ]", root);
+    JSONTEST_ASSERT(!ok);
+    JSONTEST_ASSERT(reader.getFormattedErrorMessages() ==
+                    "* Line 1, Column 3\n"
+                    "  additional six characters expected to "
+                    "parse unicode surrogate pair.\n"
+                    "See Line 1, Column 10 for detail.\n");
+  }
+  {
+    bool ok = reader.parse("[ \"\\uD801\\d1234\" ]", root);
```
Comment (BillyDonahue) [inline src/test_lib_json/main.cpp]: lowercase hex for the \u digits.

File: src/test_lib_json/main.cpp (hunk 1 above)
Comment (dota17) [reply src/test_lib_json/main.cpp]: I think getStructuredErrors() is not necessary for every testcase. Removing structured error verifications makes the testcase more clear and is convenient to add new testcase. In line 2819, we keep one structured error verifications for offset test.

Comment (BillyDonahue) [reply src/test_lib_json/main.cpp]: let's keep structured error checking and remove formatted error checking, which has less information anyway.
The structured errors can be checked by a helper function so we can make the tests compact.

**Comment by BillyDonahue** [comment]
Suggested changes submitted as PR to your fork. https://github.com/dota17/jsoncpp/pull/75

**Pull request merged**